### ISSUE TITLE
Api support for Vlan / Interfaces

### ISSUE THREAD

**Issue by tomp736** [opnsense/core#5584]
Following up from #4355 which was automatically closed last year.

interfaces_assign.php

![image](https://user-images.githubusercontent.com/12602834/154696971-a6c9438f-6e96-46f3-9c7a-1ca610ae808b.png)

I am currently limited by the lack of an API surrounding interfaces. I am working on a tool to configure wireguard and adding wg interfaces is a part of this. I am still reviewing other areas that I would like to use but have stopped short here.

- [ ] I would like to be able to list existing interfaces using an API call.
- [ ] I would like to be able to add interfaces using an API call.
- [ ] I would like to be able to delete interfaces using an API call.
- [ ] I would like to be able to update interfaces using an API call. 

I have considered modifying and importing XML configs in backup/restore config area and even using an http client that would post using webforms (as was done here https://github.com/kradalby/opnsense-go) but am not interested in going those routes as it would not be reliable enough.

I would be open to contributing to get this area added if needed :)

**Comment by fichtner** [comment]
API support for VLANs added in 22.1.4. Other interface code is subject to future change but there is no schedule / continuous funding of the effort.

**Comment by discostur** [comment]
@fichtner we are also looking for a way to assign interfaces and set IP config via API. Can you tell us if you working on it? Is it possible to create merge requests regarding interface configuration?

**Comment by fichtner** [comment]
Assignment and configuration spans probably 5000 lines of code or more and I don't see an easy way to add something in between other than reworking it all which is currently not planned. The roadmap for 23.1 hasn't been decided on, but likely due to lack of sponsoring there won't be large reworks as interface API/MVC but rather only smaller subsystems like individual interfaces or service integrations.


Cheers,
Franco

**Comment by recanman** [comment]
Have there been any updates on this? I'm using Ansible along with NetBox to configure multiple instances of OPNsense, but I'm disappointed to find out there is no API to configure interfaces. I can only create/read/update VLANs and VXLANs, but I cannot assign them.

**Comment by recanman** [comment]
I am trying to use the web form at `/interfaces_assign.php`. I was able to get this to work as the body with method being `POST` and `Content-Type: application/x-www-form-urlencoded`:
`U0NxNUFkTTZlYjVyWlpkazVkczFSQT09=bXY0c2Rxb3p1UVlXWnBXUnFzR1lSdz09&if_add={{ interface_id }}&new_entry_descr={{ interface_name }}&add_x=1`

(e.g `interface_id` is `vlan01`, `interface_name` is `vlantest`.)

What is the `U0NxNUFkTTZlYjVyWlpkazVkczFSQT09=bXY0c2Rxb3p1UVlXWnBXUnFzR1lSdz09`? Also, how would I do this from `curl`? If I do this from a Firefox-based browser, then it works (I click `Edit and Resend`). When I try to do it from `curl`, I get a CSRF error, even though I have attached a CSRF token. (header `X-CSRFToken`). I don't see a CSRF token in the headers in the web browser though...
